homeassistant/components/mqtt/models.py

```diff
@@ -231,11 +231,23 @@ def async_render_with_possible_json_value(
                 values,
                 self._value_template,
             )
-            rendered_payload = (
-                self._value_template.async_render_with_possible_json_value(
-                    payload, variables=values
+            try:
+                rendered_payload = (
+                    self._value_template.async_render_with_possible_json_value(
+                        payload, variables=values
+                    )
                 )
-            )
+            except Exception as ex:  # pylint: disable=broad-except
+                _LOGGER.error(
+                    "%s: %s rendering template for entity '%s', template: "
+                    "'%s' with payload: %s",
+                    type(ex).__name__,
+                    ex,
+                    self._entity.entity_id if self._entity else "n/a",
+                    self._value_template.template,
+                    payload,
+                )
```

Comment:
Should we just skip the payload here, since it will also be logged by `mqtt.client.async_subscribe`?
```suggestion
                _LOGGER.error(
                    "%s: %s rendering template for entity '%s', template: "
                    "'%s'",
                    type(ex).__name__,
                    ex,
                    self._entity.entity_id if self._entity else "n/a",
                    self._value_template.template,
                )
```